R/google_map_layer_parameters.R

```diff
@@ -134,7 +135,7 @@ markerColourIconCheck <- function(data, objArgs, colour, marker_icon){
   }
 
   if(!is.null(colour)){
-    if(!all((tolower(data[, colour])) %in% c("red","blue","green","lavender"))){
+    if(!all((tolower(data[["colour"]])) %in% c("red","blue","green","lavender"))){
```

Comment:
I think this should actually be `data[[colour]]` - without the quotes. Because otherwise it's looking for a column called "colour".

Reply:
Yea that is what it is doing.  The column in the `data` data frame is called "colour".

Reply:
Can you give an example which generates the error you're seeing?

For me this works
```r
library(googleway)

set_key(secret::get_secret("GOOGLE"))

df <- tram_stops
df$colour <- sample(c("red","blue","lavender"), size = nrow(df), replace = T)

google_map() %>%
  add_markers(
    data = df
    , colour = "colour"
  )

```

Reply:
The docs for the colour argument state: "string specifying the column containing the 'colour' to use for the markers. One of 'red', 'blue', 'green' or 'lavender'"

If you pass "red", "blue", or "levendar" to the colour argument you get an error.

Reply:
The string you pass to the `colour` argument must specify the **column** of the `data`.

Reply:
Ok -- If that is how the argument should work, IMO it would be more clear to document it something like "the string 'colour' if you are setting the colour of the marker or `NULL` to use the default marker colour".  I made this PR because a client of mine was confused by the docs.    

In general though, don't you want to actually pass the colour to the colour argument, not the string "colour"?  By passing the string "colour" you can't actually set the colour.

Reply:
The string 'colour' is just what I named the column of the data. You can name the column of data anything you like.

The `colour` argument works in the same way as the `lon`, `lat`, and many of the others; you use them to tell the the function which **column** of data holds that variable.

Here's another example where the column `my_column` holds the marker colours.

```r
library(googleway)

set_key(secret::get_secret("GOOGLE"))

df <- tram_stops
df$my_column <- sample(c("red","blue","lavender"), size = nrow(df), replace = T)

google_map() %>%
  add_markers(
    data = df
    , lon = "stop_lon" ## column name
    , lat = "stop_lat" ## column name
    , colour = "my_column" ## column name
  )
```